Pcap++/header/PcapLiveDeviceList.h

```diff
@@ -49,50 +57,105 @@ namespace pcpp
 
 		/**
 		 * @return A vector containing pointers to all live devices currently installed on the machine
+		 * @deprecated This method is deprecated in favor of the SmartPtrAPI overload.
+		 */
+		PCPP_DEPRECATED_RAW_PTR_API const std::vector<PcapLiveDevice*>& getPcapLiveDevicesList() const;
```

Comment:
`PcapLiveDeviceList` provides a getter API to get all registered devices, while `PcapRemoteDeviceList` provides a container like API to get all registered devices. 

In my opinion, it would be best if both lists provided a standardized API. Which one would be better?
Personally, I prefer the iterator approach, although it will be a bit trickier to implement as I dislike returning reference getters, as they make the underlying implementation difficult to change in the future.

Reply:
I think it makes sense to have both a getter and an iterator for a container-like object. 

Regarding the iterator type, since `std::shared_ptr<T>` and `T*` are pointer types, the users' existing code (such as `->`) should not have much impact.

However, if we really don't want to break the API, as you said, it's still possible to have some conversion, but it's just troublesome.

Reply:
> I think it makes sense to have both a getter and an iterator for a container-like object.

Both is an option. 
 
> Regarding the iterator type, since `std::shared_ptr<T>` and `T*` are pointer types, the users' existing code (such as `->`) should not have much impact.

Yeah, they shouldn't really. unless they decided to save the value from the iterator somewhere. That is a breaking change, but it would only need a `get()` call from the smart pointer to fix, as old code should still keep the list alive to keep the devices alive.

> However, if we really don't want to break the API, as you said, it's still possible to have some conversion, but it's just troublesome.

I tried my hand at converting iterator a bit earlier. It is technically possible but it quickly becomes templated black magic.

Reply:
I prefer to remove the old API, and have both getter and new iterator. Then we can also remove the hack of `SmartPtrApiTag` and `m_LiveDeviceListView`

@seladb  what do you say?